modules/tinymce/src/core/main/ts/fmt/FontInfo.ts

```diff
@@ -44,18 +72,33 @@ const normalizeFontFamily = (fontFamily: string) =>
   // 'Font name', Font -> Font name,Font
   fontFamily.replace(/[\'\"\\]/g, '').replace(/,\s+/g, ',');
 
-const getComputedFontProp = (propName: string, elm: HTMLElement): Optional<string> => Optional.from(DOMUtils.DOM.getStyle(elm, propName, true));
+const getSpecifiedFontProp = (propName: FontProp, elm: SugarElement): Optional<string> =>
+  Css.getRaw(elm, propName).orThunk(() => {
+    if (SugarNode.name(elm) === 'font') {
+      return Obj.get(legacyPropNames, propName).bind((legacyPropName) => Attribute.getOpt(elm, legacyPropName));
+    } else {
+      return Optional.none();
+    }
+  });
+
+const getComputedFontProp = (propName: FontProp, elm: Element): Optional<string> =>
+  Optional.from(DOMUtils.DOM.getStyle(elm, propName, true));
 
-const getFontProp = (propName: string) => (rootElm: Element, elm: Node): string => Optional.from(elm)
-  .map(SugarElement.fromDom)
-  .filter(SugarNode.isElement)
-  .bind((element: any) => getSpecifiedFontProp(propName, rootElm, element.dom)
-    .or(getComputedFontProp(propName, element.dom)))
-  .getOr('');
+const getFontInfo = (propName: FontProp, normalize: (str: string) => string) => (rootNode: Element, rngOrNode: Range | Node) => {
+  const fontProps = Arr.map(collect(DOMUtils.DOM, rngOrNode, rootNode, propName), normalize);
+  // If there is more than one font prop found, cannot declare that there is a single font
+  if (fontProps.length === 0 || fontProps.length > 1) {
+    return '';
+  } else {
+    return fontProps[0];
+  }
+};
 
-const getFontSize = getFontProp('font-size');
+const getFontSize =
```

Comment:
This is a pretty weird code style. Why was this split over 2 lines? I could understand if it was the internals of a function block, but not for an assignment.